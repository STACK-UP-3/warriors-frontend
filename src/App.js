import React from 'react';
import { Provider } from 'react-redux';
import AppRouter from './routes/index';
import configureStore from './redux/configureStore';

// Initialise the Redux store - where all application state is kept
const store = configureStore();

// Providing access to the store to every route,
// Resource:
// - https://react-redux.js.org/introduction/quick-start
// - https://redux.js.org/advanced/usage-with-react-router

const App = () => (
  <Provider store={store}>
    <AppRouter />
  </Provider>
);

export default App;
