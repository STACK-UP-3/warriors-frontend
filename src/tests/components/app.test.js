import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from '../../App';

describe('Test <App/> Component', () => {
  const component = shallow(<App />);
  it('Should render App', () => {
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
