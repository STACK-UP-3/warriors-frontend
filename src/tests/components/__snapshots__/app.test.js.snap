// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <App/> Component Should render App 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <AppRouter />
</Provider>
`;
